### ISSUE TITLE
Ensure rules are properly formatted when submitted

### ISSUE THREAD

**Issue by nagem** [scitran/core#959]
Currently, rules are not validated when submitted to the API, but are expected to be valid and complete when used by the system. For example, a user can add a rule with a match type but not a match value, which fails a key access in the rule evaluation code. 

Proposal - add a proper json schema if one does not exist or is incomplete. Validate POST and PUT payloads on project rule endpoints. Respond with a `400` or allow the validator to raise an `InputValidationException`. 

Stack trace:
```
  File "/usr/local/lib/python2.7/dist-packages/webapp2.py", line 570, in dispatch
    return method(*args, **kwargs)
  File "./api/upload.py", line 188, in upload
    return process_upload(self.request, strategy, origin=self.origin, context=context)
  File "./api/upload.py", line 137, in process_upload
    placer.process_file_field(field, file_attrs)
  File "./api/placer.py", line 186, in process_file_field
    self.save_file(field, file_attrs)
  File "./api/placer.py", line 98, in save_file
    rules.create_jobs(config.db, container_before, self.container, self.container_type)
  File "./api/jobs/rules.py", line 220, in create_jobs
    jobs_before.extend(create_potential_jobs(db, container_before, container_type, f))
  File "./api/jobs/rules.py", line 181, in create_potential_jobs
    if eval_rule(rule, file_, container):
  File "./api/jobs/rules.py", line 132, in eval_rule
    if not eval_match(match['type'], match['value'], file_, container):
KeyError: 'value'
```

**Comment by nagem** [comment]
@tcbtcb ran into this issue in the wild and it prevented all rules from evaluating if there was one rule that was incorrectly specified. The frontend client also allowed this rule to be created. 

FYI @kofalt

**Comment by kofalt** [comment]
SGTM

**Comment by ambrussimon** [comment]
Solved via #982
